managed/src/main/java/com/yugabyte/yw/commissioner/tasks/DestroyKubernetesUniverse.java

```diff
@@ -79,23 +84,44 @@ public void run() {
 
         Map<String, String> config = entry.getValue();
 
-        if (runHelmDelete) {
+        String namespace = config.get("KUBENAMESPACE");
+
+        if (runHelmDelete || namespace != null) {
           // Delete the helm deployments.
           helmDeletes.addTask(createDestroyKubernetesTask(
               universe.getUniverseDetails().nodePrefix, azName, config,
               KubernetesCommandExecutor.CommandType.HELM_DELETE,
               providerUUID));
         }
 
-        // Delete the namespaces of the deployments.
-        namespaceDeletes.addTask(createDestroyKubernetesTask(
-            universe.getUniverseDetails().nodePrefix, azName, config,
-            KubernetesCommandExecutor.CommandType.NAMESPACE_DELETE,
-            providerUUID));
+        // We depend on the fact that, deleting the namespace will
+        // delete the pull secret as well as the volumes. That won't
+        // be the case with providers which have KUBENAMESPACE
+        // paramter in the AZ config. We don't delete the namespace,
+        // as it is not created by us.
+        if (namespace != null) {
+          // Delete the PVCs created for this az.
+          volumeDeletes.addTask(createDestroyKubernetesTask(
```

Comment:
can we do this whether it is namespace based or otherwise? it is always good to delete the volumes right?

Reply:
Yep, that is correct, it is always good to clean up what we created. I have made that change.